src/main/java/org/opensearch/security/dlic/rest/api/RestApiAdminPrivilegesEvaluator.java

```diff
@@ -0,0 +1,160 @@
+            REST_API_PERMISSION_PREFIX + "/%s";
+
+    private final static String REST_ENDPOINT_ACTION_PERMISSION_PATTERN =
+            REST_API_PERMISSION_PREFIX + "/%s/%s";
+
+    private final static WildcardMatcher REST_API_PERMISSION_PREFIX_MATCHER =
+            WildcardMatcher.from(REST_API_PERMISSION_PREFIX + "/*");
+
+    @FunctionalInterface
+    public interface PermissionBuilder {
+
+        default String build() {
+            return build(null);
+        }
+
+        String build(final String action);
+
+    }
+
+    public final static Map<Endpoint, PermissionBuilder> ENDPOINTS_WITH_PERMISSIONS =
+            ImmutableMap.<Endpoint, PermissionBuilder>builder()
+                    .put(Endpoint.ACTIONGROUPS, action -> buildEndpointPermission(Endpoint.ACTIONGROUPS))
+                    .put(Endpoint.ALLOWLIST, action -> buildEndpointPermission(Endpoint.ALLOWLIST))
+                    .put(Endpoint.INTERNALUSERS, action -> buildEndpointPermission(Endpoint.INTERNALUSERS))
+                    .put(Endpoint.NODESDN, action -> buildEndpointPermission(Endpoint.NODESDN))
+                    .put(Endpoint.ROLES, action -> buildEndpointPermission(Endpoint.ROLES))
+                    .put(Endpoint.ROLESMAPPING, action -> buildEndpointPermission(Endpoint.ROLESMAPPING))
+                    .put(Endpoint.TENANTS, action -> buildEndpointPermission(Endpoint.TENANTS))
+                    .put(Endpoint.SSL, action -> {
+                        switch (action) {
+                            case "certs":
+                                return buildEndpointActionPermission(Endpoint.SSL, "certs/info");
+                            case "reloadcerts":
+                                return buildEndpointActionPermission(Endpoint.SSL, "certs/reload");
+                            default:
+                                return null;
+                        }
+                    }).build();
+
+    private final ThreadContext threadContext;
+
+    private final PrivilegesEvaluator privilegesEvaluator;
+
+    private final AdminDNs adminDNs;
+
+    public RestApiAdminPrivilegesEvaluator(
+            final ThreadContext threadContext,
+            final PrivilegesEvaluator privilegesEvaluator,
+            final AdminDNs adminDNs) {
+        this.threadContext = threadContext;
+        this.privilegesEvaluator = privilegesEvaluator;
+        this.adminDNs = adminDNs;
+    }
+
+    public boolean isCurrentUserRestApiAdminFor(final Endpoint endpoint, final String action) {
+        final Pair<User, TransportAddress> userAndRemoteAddress = Utils.userAndRemoteAddressFrom(threadContext);
+        if (userAndRemoteAddress.getLeft() == null) {
+            return false;
+        }
+        if (adminDNs.isAdmin(userAndRemoteAddress.getLeft())) {
+            if (logger.isDebugEnabled()) {
+                logger.debug(
+                        "Security admin permissions required for endpoint {} but {} is not an admin",
```

Comment:
This log message is the opposite of what it should be. The original block in `RestApiPrivilegesEvaluator` was:

```
if (!adminDNs.isAdminDN(sslInfo.getPrincipal())) {
	logger.warn("Security admin permissions required but {} is not an admin", sslInfo.getPrincipal());
	return "Security admin permissions required but " + sslInfo.getPrincipal() + " is not an admin";
}
```

if the user is connecting with admin dn then they are said to be super admin and have privileges to perform all actions in the cluster.

Reply:
ok will change np.

Reply:
Ahhh sorry misread the comment. The code in the master:
```java
User user = threadPool.getThreadContext().getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
return adminDNs.isAdmin(user);
```
I only added a debug message. The purpose of `isSuperAdmin()` stays the same. `RestApiAdminPrivilegesEvaluator` just adds additional check for the permission  nothing more since the logged in user can has such permissions to change e.g. `hidden` or `reserved` fields. Maybe the name for the class  `RestApiAdminPrivilegesEvaluator` needs to be changed for better understanding what it does. WDYT?